File: VB.NET/GetSizeForm.vb
Imports System.Globalization
Imports System.Windows.Forms
Imports Vintasoft.Barcode

Public Partial Class GetSizeForm
	Inherits Form

	#Region "Constructors"

	Public Sub New(variable As String, value As Single, dpi As Integer, units As UnitOfMeasure)
		InitializeComponent()
		cbUnits.Items.Add(UnitOfMeasure.Pixels)
		cbUnits.Items.Add(UnitOfMeasure.Inches)
		cbUnits.Items.Add(UnitOfMeasure.Centimeters)
		cbUnits.Items.Add(UnitOfMeasure.Millimeters)
		cbUnits.SelectedItem = units
		variableValue.Text = value.ToString(CultureInfo.InvariantCulture)
		dpiValue.Value = dpi
		Text = String.Format(Text, variable)
		labelSize.Text = String.Format(labelSize.Text, variable)
		_dpi = dpi
		_value = value
		_units = units
	End Sub



	#End Region



	#Region "Properties"

	Private _value As Single
	Friend ReadOnly Property Value() As Single
		Get
			Return _value
		End Get
	End Property

	Private _dpi As Integer
	Friend ReadOnly Property Resolution() As Integer
		Get
			Return _dpi
		End Get
	End Property

	Private _units As UnitOfMeasure
	Friend ReadOnly Property Units() As UnitOfMeasure
		Get
			Return _units
		End Get
	End Property

	#End Region



	#Region "Methods"

    Private Sub buttonOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonOK.Click
        _units = DirectCast(cbUnits.SelectedItem, UnitOfMeasure)
        _dpi = CInt(Math.Truncate(dpiValue.Value))
        _value = Single.Parse(variableValue.Text.Replace(","c, "."c), CultureInfo.InvariantCulture)
        DialogResult = DialogResult.OK
        Close()
    End Sub

    Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonCancel.Click
        DialogResult = DialogResult.Cancel
        Close()
    End Sub

	#End Region

End Class

File: CSharp/GetSizeForm.cs
﻿using System;
using System.Globalization;
using System.Windows.Forms;

using Vintasoft.Barcode;

namespace BarcodeDemo
{
    /// <summary>
    /// A form that allows to set size of barcode image.
    /// </summary>
    public partial class GetSizeForm : Form
    {

        #region Constructors

        /// <summary>
        /// Initializes a new instance of the <see cref="GetSizeForm"/> class.
        /// </summary>
        public GetSizeForm()
        {
            InitializeComponent();
            unitsValueEditor.Items.Add(UnitOfMeasure.Pixels);
            unitsValueEditor.Items.Add(UnitOfMeasure.Inches);
            unitsValueEditor.Items.Add(UnitOfMeasure.Centimeters);
            unitsValueEditor.Items.Add(UnitOfMeasure.Millimeters);
            unitsValueEditor.SelectedItem = UnitOfMeasure.Pixels;
        }

        #endregion



        #region Properties

        /// <summary>
        /// Gets or sets an image width, in units.
        /// </summary>
        public float WidthValue
        {
            get
            {
                string value = widthValueEditor.Text.Replace(',', '.');
                return float.Parse(value, CultureInfo.InvariantCulture);
            }
            set
            {
                widthValueEditor.Text = value.ToString(CultureInfo.InvariantCulture);
            }
        }

        /// <summary>
        /// Gets or sets an image height, in units.
        /// </summary>
        public float HeightValue
        {
            get
            {
                string value = heightValueEditor.Text.Replace(',', '.');
                return float.Parse(value, CultureInfo.InvariantCulture);
            }
            set
            {
                heightValueEditor.Text = value.ToString(CultureInfo.InvariantCulture);
            }
        }

        /// <summary>
        /// Gets or sets the unit of measure of width and height.
        /// </summary>
        public UnitOfMeasure UnitsValue
        {
            get
            {
                return (UnitOfMeasure)unitsValueEditor.SelectedItem;
            }
            set
            {
                unitsValueEditor.SelectedItem = value;
            }
        }

        /// <summary>
        /// Gets or sets an image resolution.
        /// </summary>
        public float ResolutionValue
        {
            get
            {
                return (float)resolutionValueEditor.Value;
            }
            set
            {
                resolutionValueEditor.Value = (int)Math.Round(value);
            }
        }

        #endregion



        #region Methods

        /// <summary>
        /// Closes this dialog with "OK" dialog result.
        /// </summary>
        private void buttonOK_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        /// <summary>
        /// Closes this dialog with "Cancel" dialog result.
        /// </summary>
        private void buttonCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        #endregion

    }
}